streamalert/alert_processor/outputs/credentials/provider.py

```diff
@@ -331,134 +333,84 @@ def get_formatted_output_credentials_name(service_name, descriptor):
+        """Always returns True, as SSM is the place where all encrypted credentials are
            guaranteed to be cold-stored."""
         return True
 
-    def save_credentials_into_s3(self, descriptor, credentials, kms_key_alias):
-        """Takes the given credentials, encrypts them, and saves them to AWS S3.
-
-        Notably, this implementation is NOT for the CredentialsCachingDriver interface, as the
-        S3Driver is not a caching driver.
+    def save_credentials(self, descriptor, credentials, kms_key_alias):
+        """Takes the given credentials and saves them to AWS SSM Parameter store.
 
         Args:
             descriptor (str): Descriptor of the current Output
-            credentials (Credentials): Credentials object to be saved into S3
+            credentials (Credentials): Credentials object to be saved into SSM Param Store
             kms_key_alias (str): KMS key alias for streamalert secrets
 
         Returns:
             bool: True on success, False otherwise.
         """
-        s3_key = get_formatted_output_credentials_name(self._service_name, descriptor)
-
-        # Encrypt the creds and push them to S3
-        if not credentials.is_encrypted():
-            credentials.encrypt(self._region, kms_key_alias)
-
-        encrypted_credentials = credentials.data()
-        if not encrypted_credentials:
-            return True
-
-        try:
-            return AwsS3.put_object(
-                encrypted_credentials,
-                bucket=self._bucket,
-                key=s3_key,
-                region=self._region
-            )
-        except ClientError:
-            LOGGER.exception(
-                'An error occurred while sending credentials to S3 for key \'%s\' in bucket \'%s\'',
-                s3_key,
-                self._bucket
-            )
-            return False
+        # If the creds are encrypted, decrypt them as SSM will handle the encryption
+        unencrypted_creds = (
+            credentials.get_data_kms_decrypted()
+            if credentials.is_encrypted()
+            else credentials.data()
+        )
 
-    def get_s3_key(self, descriptor):
-        """Returns an appropriate S3 bucket key for credentials relevant to this Output.
+        parameter_name = self.get_parameter_name(descriptor)
 
-        Args:
-            descriptor (str): Descriptor of the current Output
+        return AwsSsm.put_parameter(parameter_name, unencrypted_creds, self._region, kms_key_alias)
 
-        Returns:
-            string
-        """
-        return get_formatted_output_credentials_name(self._service_name, descriptor)
-
-    def get_s3_secrets_bucket(self):
-        """Returns an appropriate S3 bucket for all credentials relevant to this driver.
+    def get_parameter_prefix(self):
+        """Generate the parameter prefix from prefix and descriptor"""
+        return "{}_streamalert_secrets".format(self._prefix)
```

Comment:
can we update this to be something a little more specific to `outputs`? right now, apps also use ssm (even though they are not very well implemented to use `paths`..)

I'm thinking a namespacing convention like:

`/streamalert/outputs/<output_type>/<output_paramter_suffix>`

where the end result would be something like:

``/streamalert/outputs/slack/my-cool-output`

... might be ideal here, since SSM allows for getting params by "path".

Thoughts? @Ryxias ?

Reply:
Agree wholeheartedly; but what about Prefix?

Seems like the namespace convention is:

`/[PREFIX]/[APPLICATION]/[COMPONENT]/[DRIVER_OR_TYPE]/[INSTANCE]`

Reply:
If you look at line 410, this is how the parameter name is generated. Sorry if this isn't clear

Reply:
sorry I didn't see your response and just edited for prefix. unsure tbh

Reply:
@jack1902 it was clear how it's generated - but I'm suggesting we use a different convention than that :)